File: Android-Java/app/src/main/java/io/supernova/spacebook/adapter/ProfilePhotosActivityGalleryRecyclerViewAdapter.java
/**
 *  Created by Supernova.
 */

package io.supernova.spacebook.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;
import java.util.*;
import android.text.Spannable;
import android.view.LayoutInflater;
import android.text.style.ForegroundColorSpan;
import io.supernova.spacebook.R;
import android.text.style.RelativeSizeSpan;
import android.widget.TextView;
import android.view.View;
import android.text.SpannableString;
import android.widget.ImageView;


public class ProfilePhotosActivityGalleryRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    public static final int TITLE_VIEW_HOLDER_VIEW_TYPE = 1;
    public static final int VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE = 2;
    public static final int VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE = 3;
    public static final int TITLE_TWO_VIEW_HOLDER_VIEW_TYPE = 4;
    public static final int VIEW_SIX_VIEW_HOLDER_VIEW_TYPE = 5;
    public static final int VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE = 6;
    public static final int VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE = 7;
    
    public static final List<Integer> MOCK_DATA = Arrays.asList(TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE, TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE, TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE);
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    
        switch (viewType) {
            case TITLE_VIEW_HOLDER_VIEW_TYPE: 
                return new TitleViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.title_view_holder, parent, false));
            case VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE: 
                return new ViewFourViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_four_view_holder, parent, false));
            case VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE: 
                return new ViewFiveViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_five_view_holder, parent, false));
            case TITLE_TWO_VIEW_HOLDER_VIEW_TYPE: 
                return new TitleTwoViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.title_two_view_holder, parent, false));
            case VIEW_SIX_VIEW_HOLDER_VIEW_TYPE: 
                return new ViewSixViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_six_view_holder, parent, false));
            case VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE: 
                return new ViewSevenViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_seven_view_holder, parent, false));
            case VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE: 
                return new ViewEightViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.view_eight_view_holder, parent, false));
        }
        
        throw new RuntimeException("Unsupported view type");
    }
    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
    
        // Here you can bind RecyclerView item data.
    }
    @Override
    public int getItemViewType(int position) {
    
        return MOCK_DATA.get(position);
    }
    @Override
    public int getItemCount() {
    
        return MOCK_DATA.size();
    }
    
    
    public static class TitleViewHolder extends RecyclerView.ViewHolder {
        private TextView todayTextView;
        public TitleViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
            // Configure Today component
            todayTextView = itemView.findViewById(R.id.today_text_view);
            SpannableString todayTextViewText = new SpannableString(itemView.getContext().getString(R.string.title_view_holder_today_text_view_text));
            todayTextView.setText(todayTextViewText);
        }
    }
    
    
    public static class ViewFourViewHolder extends RecyclerView.ViewHolder {
        public ViewFourViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
        }
    }
    
    
    public static class ViewFiveViewHolder extends RecyclerView.ViewHolder {
        public ViewFiveViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
        }
    }
    
    
    public static class TitleTwoViewHolder extends RecyclerView.ViewHolder {
        private TextView tomorrowTextView;
        public TitleTwoViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
            // Configure Tomorrow component
            tomorrowTextView = itemView.findViewById(R.id.tomorrow_text_view);
            SpannableString tomorrowTextViewText = new SpannableString(itemView.getContext().getString(R.string.title_two_view_holder_tomorrow_text_view_text));
            tomorrowTextView.setText(tomorrowTextViewText);
        }
    }
    
    
    public static class ViewSixViewHolder extends RecyclerView.ViewHolder {
        public ViewSixViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
        }
    }
    
    
    public static class ViewSevenViewHolder extends RecyclerView.ViewHolder {
        public ViewSevenViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
        }
    }
    
    
    public static class ViewEightViewHolder extends RecyclerView.ViewHolder {
        public ViewEightViewHolder(View itemView) {
            super(itemView);
            init();
        }
        public void init() {
        
        }
    }
}

File: Android-Kotlin/app/src/main/java/io/supernova/spacebook/adapter/ProfilePhotosActivityGalleryRecyclerViewAdapter.kt
/**
 * Created by Supernova.
 */

package io.supernova.spacebook.adapter

import android.view.ViewGroup
import android.support.v7.widget.RecyclerView
import io.supernova.spacebook.databinding.TitleTwoViewHolderBinding
import java.util.*
import io.supernova.spacebook.databinding.ViewSevenViewHolderBinding
import io.supernova.spacebook.databinding.ViewSixViewHolderBinding
import io.supernova.spacebook.databinding.ViewFourViewHolderBinding
import io.supernova.spacebook.databinding.ViewEightViewHolderBinding
import io.supernova.spacebook.databinding.TitleViewHolderBinding
import io.supernova.spacebook.databinding.ViewFiveViewHolderBinding
import android.view.LayoutInflater
import io.supernova.spacebook.R
import android.databinding.DataBindingUtil
import android.view.View


class ProfilePhotosActivityGalleryRecyclerViewAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>() {

	companion object {
		val TITLE_VIEW_HOLDER_VIEW_TYPE = 1
		val VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE = 2
		val VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE = 3
		val TITLE_TWO_VIEW_HOLDER_VIEW_TYPE = 4
		val VIEW_SIX_VIEW_HOLDER_VIEW_TYPE = 5
		val VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE = 6
		val VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE = 7
		
		val MOCK_DATA = listOf(TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE, TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE, TITLE_VIEW_HOLDER_VIEW_TYPE, VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE, VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE, TITLE_TWO_VIEW_HOLDER_VIEW_TYPE, VIEW_SIX_VIEW_HOLDER_VIEW_TYPE, VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE, VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE)
	}
	
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
	
		when (viewType) {
			TITLE_VIEW_HOLDER_VIEW_TYPE -> {
				return TitleViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.title_view_holder, parent, false))
			}
			VIEW_FOUR_VIEW_HOLDER_VIEW_TYPE -> {
				return ViewFourViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_four_view_holder, parent, false))
			}
			VIEW_FIVE_VIEW_HOLDER_VIEW_TYPE -> {
				return ViewFiveViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_five_view_holder, parent, false))
			}
			TITLE_TWO_VIEW_HOLDER_VIEW_TYPE -> {
				return TitleTwoViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.title_two_view_holder, parent, false))
			}
			VIEW_SIX_VIEW_HOLDER_VIEW_TYPE -> {
				return ViewSixViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_six_view_holder, parent, false))
			}
			VIEW_SEVEN_VIEW_HOLDER_VIEW_TYPE -> {
				return ViewSevenViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_seven_view_holder, parent, false))
			}
			VIEW_EIGHT_VIEW_HOLDER_VIEW_TYPE -> {
				return ViewEightViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.view_eight_view_holder, parent, false))
			}
		}
		
		throw RuntimeException("Unsupported view type")
	}
	
	override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
	
		// Here you can bind RecyclerView item data.
	}
	
	override fun getItemViewType(position: Int): Int {
	
		return MOCK_DATA.get(position)
	}
	
	override fun getItemCount(): Int {
	
		return MOCK_DATA.size
	}
	
	
	class TitleViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: TitleViewHolderBinding
		init {
			binding = TitleViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class ViewFourViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: ViewFourViewHolderBinding
		init {
			binding = ViewFourViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class ViewFiveViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: ViewFiveViewHolderBinding
		init {
			binding = ViewFiveViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class TitleTwoViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: TitleTwoViewHolderBinding
		init {
			binding = TitleTwoViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class ViewSixViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: ViewSixViewHolderBinding
		init {
			binding = ViewSixViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class ViewSevenViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: ViewSevenViewHolderBinding
		init {
			binding = ViewSevenViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
	
	
	class ViewEightViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
	
		private lateinit var binding: ViewEightViewHolderBinding
		init {
			binding = ViewEightViewHolderBinding.bind(itemView)
			init()
		}
		
		fun init() {
		
		}
	}
}
